airflow/providers/amazon/aws/notifications/sns.py

```diff
@@ -0,0 +1,94 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from functools import cached_property
+from typing import Sequence
+
+from airflow.exceptions import AirflowOptionalProviderFeatureException
+from airflow.providers.amazon.aws.hooks.sns import SnsHook
+
+try:
+    from airflow.notifications.basenotifier import BaseNotifier
+except ImportError:
+    raise AirflowOptionalProviderFeatureException(
+        "Failed to import BaseNotifier. This feature is only available in Airflow versions >= 2.6.0"
+    )
+
+
+class SnsNotifier(BaseNotifier):
+    """
+    Amazon SNS (Simple Notification Service) Notifier.
+
+    .. seealso::
+        For more information on how to use this notifier, take a look at the guide:
+        :ref:`howto/notifier:SnsNotifier`
+
+    :param aws_conn_id: The :ref:`Amazon Web Services Connection id <howto/connection:aws>`
+        used for AWS credentials. If this is None or empty then the default boto3 behaviour is used.
+    :param target_arn: either a TopicArn or an EndpointArn.
+    :param message: The message you want to send.
+    :param subject: The message subject you want to send.
+    :param message_attributes: The message attributes you want to send as a flat dict (data type will be
+        determined automatically).
+    :param region_name: AWS region_name. If not specified then the default boto3 behaviour is used.
+    """
+
+    template_fields: Sequence[str] = (
+        "target_arn",
+        "message",
+        "subject",
+        "message_attributes",
+        "aws_conn_id",
+        "region_name",
+    )
+
+    def __init__(
+        self,
+        *,
+        aws_conn_id: str | None = SnsHook.default_conn_name,
+        target_arn: str,
+        message: str,
+        subject: str | None = None,
+        message_attributes: dict | None = None,
+        region_name: str | None = None,
+    ):
+        super().__init__()
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.target_arn = target_arn
+        self.message = message
+        self.subject = subject
+        self.message_attributes = message_attributes
+
+    @cached_property
+    def hook(self) -> SnsHook:
+        """Amazon SNS Hook (cached)."""
+        return SnsHook(aws_conn_id=self.aws_conn_id, region_name=self.region_name)
+
+    def notify(self, context):
+        """Publish the notification message to Amazon SNS."""
+        self.hook.publish_to_target(
+            target_arn=self.target_arn,
+            message=self.message,
+            subject=self.subject,
+            message_attributes=self.message_attributes,
+        )
+
+
+send_sns_notification = SnsNotifier
```

Comment:
Same convention in other notifiers `¯\_(ツ)_/¯`.
https://github.com/apache/airflow/blob/b82ce61285f3f4f0c7eccb2c3effaef53c9fb84e/airflow/providers/amazon/aws/notifications/chime.py#L61

https://github.com/apache/airflow/blob/b82ce61285f3f4f0c7eccb2c3effaef53c9fb84e/airflow/providers/slack/notifications/slack.py#L91

https://github.com/apache/airflow/blob/9d8c77e447f5515b9a6aa85fa72511a86a128c28/airflow/providers/apprise/notifications/apprise.py#L102

And this conventions was here since first notifier implemented, I know because this PR sits in my local machine for a long time I've just waited until 2.6 released... and have a time to push it only now.

Reply:
I am not against it but I dont feel it is necessary and bring more complexity by adding a proxy name ¯\_(ツ)_/¯

Reply:
I've tried to find why initially we decide to use `snake_case` for notifiers, but with no luck
- https://github.com/apache/airflow/issues/28405
- https://github.com/apache/airflow/pull/28569